pkg/operation/botanist/component/kubeapiserver/deployment.go

```diff
@@ -552,6 +558,19 @@ func (k *kubeAPIServer) computeKubeAPIServerCommand() []string {
 	out = append(out, "--requestheader-group-headers=X-Remote-Group")
 	out = append(out, "--requestheader-username-headers=X-Remote-User")
 
+	if k.values.IsNodeless {
+		disableAPIs := map[string]bool{
+			"autoscaling/v2":                 false,
+			"batch/v1":                       false,
+			"apps/v1":                        false,
```

Comment:
Yes, we wanted to start with basic apis and make this list exhaustive later.
>  It will be unmanageable if we have to do it. Is this disabling really required?
Should we allow the creation of these resources and let it leave to users that they should know that such resources should not be created if there is no node?